### ISSUE TITLE
Builtin util for mounting UI's in content scripts.

### ISSUE THREAD

**Issue by aklinker1** [wxt-dev/wxt#139]
### Feature Request

Use `@webext-core/isolated-element` internally. Add another option to `cssInjectionMode`, `"shadowRoot"`, that will automatically handle applying the style inside the created shadow root.

```ts
function mountContentScriptUi<T>({
  type: "inline" | "overlay" | "modal";
  overlayAlignment?: `${"top" | "bottom" | "center"}-${"left" | "right" | "center"}` | "center"
  append?: "add" | "replace" | "before" | "after" | (anchor: HTMLElement) => void;
  anchor?: string | HTMLElement | (() => string | HTMLElement)
  mount: (container: HtmlElement) => T;
  onRemove?: (mounted: T): => void;
  styles?: string;
  attrs?: Record<string, string>;
  zIndex?: number;
}): Promise<{
  shadowHost: HTMLElement,
  container: HTMLElement,
  shadow: ShadowRoot,
  mounted: T,
  mount: () => void,
  remove: () => void,
}>;
```

#### Is your feature request related to a bug?

N/A

### What are the alternatives?

N/A

### Additional context

- #138 
- https://webext-core.aklinker1.io/guide/isolated-element/

**Comment by aklinker1** [comment]
We should also add the content script CSS files to the manifest's web resources automatically.

**Comment by aklinker1** [comment]
Released in [`v0.7.1`](https://github.com/wxt-dev/wxt/releases/tag/v0.7.1)
